### DISCUSSION TITLE
Error with dpgen autotest post during relaxation

### DISCUSSION THREAD

**Discussion by Daisy315** [deepmodeling/dpgen#965] [Q&A] [answered]
Hello developers, I got an error with dpgen autotest post during relaxation, in my log.lammps there are three trajectories during minimization, and when I do _post relaxation.json_, the dispatcher only extract the last snapshot for the last two trajectories and the size of energies could not match, the output is as follows and my dpgen version is 0.10.4.

 File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/bin/dpgen", line 8, in <module>
    sys.exit(main())
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpgen/main.py", line 175, in main
    args.func(args)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpgen/auto_test/run.py", line 57, in gen_test
    run_task(args.TASK, args.PARAM, args.MACHINE)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpgen/auto_test/run.py", line 44, in run_task
    post_equi(confs, inter_parameter)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpgen/auto_test/common_equi.py", line 212, in post_equi
    res = inter.compute(ii)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpgen/auto_test/Lammps.py", line 369, in compute
    d_dump = loadfn(contcar)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/monty/serialization.py", line 72, in loadfn
    return json.load(fp, *args, **kwargs)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/json/__init__.py", line 293, in load
    return loads(fp.read(),
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/json/__init__.py", line 359, in loads
    return cls(**kw).decode(s)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/monty/json.py", line 454, in decode
    return self.process_decoded(d)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/monty/json.py", line 413, in process_decoded
    return cls_.from_dict(data)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/monty/json.py", line 180, in from_dict
    return cls(**decoded)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpdata/system.py", line 223, in __init__
    self.check_data()
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpdata/system.py", line 243, in check_data
    dd.check(self)
  File "/public1/home/sca1631/software/miniconda/envs/deepmd-2.1.3/lib/python3.9/site-packages/dpdata/system.py", line 121, in check
    raise DataError("Shape of %s is %s, but expected %s" % (self.name,
dpdata.system.DataError: Shape of energies is (25,), but expected (34,)

**_my relaxation.json:_**
{
    "structures": ["confs/std-fcc"],
    "interaction": {
    "type": "deepmd",
    "model": "graph.pb",
    "in_lammps": "lammps_input/in.lammps",
    "type_map": {"Li": 0,"P": 1,"S": 2}
},
"relaxation": {
    "cal_type":      "relaxation",
    "cal_setting":  {
                     "relax_pos":true,
                     "relax_shape":true,
                     "relax_vol":true,
					 "etol":1e-12,
					 "ftol":1e-6,
					 "maxiter":5000,
					 "maximal":500000}
}
}
**_log.lammps:_**
 Step         PotEng          Pxx            Pyy            Pzz            Pxy            Pxz            Pyz             Lx             Ly             Lz           Volume         c_mype    
         0  -184.91634     -3681.5531      543.48448      216.9544       828.76722     -768.50274     -464.97446      6.188838       12.358286      11.463493      876.76727     -184.91634    
       100  -184.96563     -1183.9799      1717.6673     -260.08288     -441.28109     -127.61554     -76.867242      6.1801736      12.340984      11.447444      873.09        -184.96563    
       200  -184.98597     -1341.9624      1632.0681     -419.13648     -644.09665     -159.22232     -219.24899      6.1801736      12.340984      11.447444      873.09        -184.98597    
       300  -185.00078     -1396.2282      1667.4345     -360.15754     -703.03199     -179.03731     -325.46711      6.1801736      12.340984      11.447444      873.09        -185.00078    
       400  -185.024       -1459.98        1822.0655     -222.76083     -725.50166     -213.45913     -467.36695      6.1801736      12.340984      11.447444      873.09        -185.024      
       500  -185.03772     -1584.7052      1889.7865     -177.59963     -723.17682     -261.56655     -527.22254      6.1801736      12.340984      11.447444      873.09        -185.03772    
       600  -185.05562     -1944.9432      1863.3266     -266.37885     -694.77103     -295.93484     -589.91282      6.1801736      12.340984      11.447444      873.09        -185.05562    
       700  -185.07701     -2174.3437      2027.6208     -209.62904     -664.8799      -237.87903     -662.4308       6.1789359      12.338512      11.445152      872.56552     -185.07701    
       800  -185.09485     -2158.6244      2195.2519     -146.46873     -691.54437     -99.763602     -716.60542      6.1776981      12.336041      11.442859      872.04124     -185.09485    
       900  -185.14078     -2056.1276      2396.985      -175.50873     -671.0037       331.20678     -840.29934      6.1743351      12.329325      11.43663       870.61787     -185.14078    
      1000  -185.15976     -1820.232       2399.4585     -225.4213      -772.09029      472.43051     -870.38415      6.1730974      12.326854      11.434337      870.09438     -185.15976    
      1100  -185.17822     -1727.8068      2174.0169     -518.84148     -912.80123      588.95013     -866.55439      6.1730974      12.326854      11.434337      870.09438     -185.17822    
      1200  -185.19347     -1646.0664      2010.2089     -724.9269      -1028.7936      672.3004      -855.80711      6.1730974      12.326854      11.434337      870.09438     -185.19347    
      1300  -185.21673     -1220.7014      2084.8439     -657.66884     -1176.8953      779.62451     -854.29759      6.1718596      12.324382      11.432045      869.5711      -185.21673    
      1400  -185.29246     -814.98081      1552.792      -1102.3445     -1537.4873      1142.8192     -725.99014      6.1706218      12.32191       11.429752      869.04802     -185.29246    
      1500  -185.31883     -325.57221      1575.027      -1002.5351     -1617.3105      1295.3517     -692.71092      6.1693841      12.319439      11.427459      868.52516     -185.31883    
      1600  -185.33327     -148.83495      1414.2191     -1130.4086     -1654.0826      1394.5336     -660.0126       6.1693841      12.319439      11.427459      868.52516     -185.33327    
      1700  -185.35025      152.48995      1236.6869     -1277.3108     -1700.0826      1531.2593     -621.69211      6.1693841      12.319439      11.427459      868.52516     -185.35025    
      1800  -185.36364      510.12333      1132.7656     -1377.378      -1735.5076      1661.3925     -595.3225       6.1693841      12.319439      11.427459      868.52516     -185.36364    
      1900  -185.38203      968.58765      887.69621     -1730.872      -1744.946       1879.9244     -567.29917      6.1706218      12.32191       11.429752      869.04802     -185.38203    
      2000  -185.41393      577.76663      1410.884      -2224.0283     -1992.8271      2157.6459     -897.59866      6.1758321      12.332315      11.439403      871.25127     -185.41393    
      2100  -185.4197       615.63376      1678.7124     -2296.1285     -2076.1686      2074.1551     -1114.5         6.1782921      12.337227      11.443959      872.29282     -185.4197     
      2112  -185.4197       614.92911      1679.883      -2295.4687     -2077.4158      2073.5246     -1115.4297      6.178297       12.337237      11.443968      872.2949      -185.4197     
Loop time of 62.7002 on 1 procs for 2112 steps with 42 atoms

   Step         PotEng          Pxx            Pyy            Pzz            Pxy            Pxz            Pyz             Lx             Ly             Lz           Volume         c_mype    
      2112  -185.4197       614.92911      1679.883      -2295.4687     -2077.4158      2073.5246     -1115.4297      6.178297       12.337237      11.443968      872.2949      -185.4197     
      2142  -185.42212      113.55548      436.84548     -433.25347     -1640.5862      1658.4968     -1891.3321      6.1819426      12.359581      11.419419      872.51464     -185.42212    
Loop time of 2.44511 on 1 procs for 30 steps with 42 atoms

   Step         PotEng          Pxx            Pyy            Pzz            Pxy            Pxz            Pyz             Lx             Ly             Lz           Volume         c_mype    
      2142  -185.42212      113.55548      436.84548     -433.25347     -1640.5862      1658.4968     -1891.3321      6.1819426      12.359581      11.419419      872.51464     -185.42212    
      2200  -185.4286       17.202644      24.449093     -52.907213     -1008.1086      1367.0718     -238.66642      6.1832812      12.366522      11.40813       872.33049     -185.4286     
      2300  -185.43503      10.223457      26.591463     -24.71836      -314.07384      277.23271     -13.023918      6.1852465      12.376651      11.398474      872.58324     -185.43503    
      2400  -185.44233      16.721636      22.214733      2.9062418     -95.801521      55.32739      -6.3682556      6.1910075      12.402797      11.375438      873.47222     -185.44233    
      2500  -185.44736      111.19575      121.276        88.268509     -30.447731      24.454398      14.726637      6.1943785      12.429508      11.350908      873.94131     -185.44736    
      2600  -185.4499       12.456544      19.799904      29.478575     -2.4052518      25.280005      62.771741      6.1937705      12.450851      11.333868      874.04201     -185.4499     
      2700  -185.45072     -2.9291052     -47.833708     -6.3393029     -1.3719618     -17.356619     -43.880772      6.192411       12.463356      11.325787      874.1041      -185.45072    
      2800  -185.4511      -61.216816     -32.146008      2.519208       4.5944188      20.900691     -21.716379      6.1914151      12.470931      11.32155       874.16755     -185.4511     
      2900  -185.4513      -2.1073671      4.6766718      2.3309892     -0.28697559     7.4814664      15.973352      6.1899858      12.476462      11.319533      874.19759     -185.4513     
      3000  -185.45135     -1.9871567      4.0101871     -2.8611581     -0.81947121     0.30785993     1.0762628      6.1893545      12.479596      11.318722      874.26533     -185.45135    
      3077  -185.45135      0.092494872    0.13862923     0.051961131   -0.011314341   -0.0072790059  -0.1142191      6.1891419      12.480738      11.318367      874.28792     -185.45135    
Loop time of 35.3068 on 1 procs for 935 steps with 42 atoms

**_energies in CONTCAR_**(it seems CONTCAR to be the transition of result.json?)
        "energies": {
            "@module": "numpy",
            "@class": "array",
            "dtype": "float64",
            "data": [
                -184.91634,
                -184.96563,
                -184.98597,
                -185.00078,
                -185.024,
                -185.03772,
                -185.05562,
                -185.07701,
                -185.09485,
                -185.14078,
                -185.15976,
                -185.17822,
                -185.19347,
                -185.21673,
                -185.29246,
                -185.31883,
                -185.33327,
                -185.35025,
                -185.36364,
                -185.38203,
                -185.41393,
                -185.4197,
                -185.4197,
                -185.42212,
                -185.45135
            ]
        },

**Comment by HuangJiameng** [comment, marked as answer]
Maybe this typo cause the error? https://github.com/deepmodeling/dpdata/pull/351 You can fix it and try again.

**Reply by Daisy315** [reply]
I tried and it didn't work. The error message said something wrong in line 121 of system.py, it sasy:
    def check(self, system: "System"):
        """Check if a system has correct data of this type.
        
        Parameters
        ----------
        system : System
            checked system
        
        Raises
        ------
        DataError
            type or shape of data is not correct
        """
        # check if exists
        if self.name in system.data:
            data = system.data[self.name]
            # check dtype
            # allow list for empty np.ndarray
            if isinstance(data, list) and not len(data):
                pass
            elif not isinstance(data, self.dtype):
                raise DataError("Type of %s is %s, but expected %s" % (self.name,
                                type(data).__name__, self.dtype.__name__))
            # check shape
            if self.shape is not None:
                shape = self.real_shape(system)
                # skip checking empty list of np.ndarray
                if isinstance(data, np.ndarray):
                    if data.size and shape != data.shape:
                        raise DataError("Shape of %s is %s, but expected %s" % (self.name,
                                data.shape, shape))

**Reply by HuangJiameng** [reply]
It seems that only the energies of the first two traj are extracted.

**Reply by Daisy315** [reply]
-185.45135 is the last one of the third traj, and I think it may origin form the sourcecode but I don't know how to fix it...

**Reply by ZLI-afk** [reply]
@Daisy315 Could you please share your latest autotest document including parameters.json with us so we can look into it and see what happened?

**Reply by Daisy315** [reply]
Of course, this is my attachment.
[relax.zip](https://github.com/deepmodeling/dpgen/files/9638562/relax.zip)

**Reply by ZLI-afk** [reply]
We have found that the content of auto-generated CONTCAR in your work direction is kind of weird (content mismatched with file name). Maybe you should double-check about that again, or you may share your potential with us so that we can test it from scratch.

**Reply by Daisy315** [reply]
So sorry that I forgot to share the potential file.
[graph.zip](https://github.com/deepmodeling/dpgen/files/9638615/graph.zip)

**Reply by ZLI-afk** [reply]
Our test with default in.lammps is passed with no error. Maybe you should try the default one and see if the problem is fixed. Please refer to our results attached.
[relax_test.zip](https://github.com/deepmodeling/dpgen/files/9638637/relax_test.zip)

**Reply by Daisy315** [reply]
Many thanks to your reply and finally I found it was mainly due to the version of deepmd-kit, version 2.1.3 not works well with dpgen autotest post(version 0.10.4) but 2.1.1 can do, and maybe this can help others in need.

**Reply by ZLI-afk** [reply]
Noticed with thanks. It may be due to some bugs corresponding to autotest. We have already identified and will fix them in the future release:)

**Comment by HuangJiameng** [comment]
@ZLI-afk Could you look at this problem? thx!
